src/main/scala/fi/oph/koski/valpas/repository/ValpasDatabaseService.scala

```diff
@@ -0,0 +1,106 @@
+  }
+
+  def getPeruskoulunValvojalleNäkyvätOppijat(oppilaitosOids: Seq[String]): Seq[ValpasOppija] =
+    db.runDbSync(sql"""
+      SELECT
+        r_henkilo.oppija_oid,
+        r_henkilo.hetu,
+        r_henkilo.syntymaaika,
+        r_henkilo.etunimet,
+        r_henkilo.sukunimi,
+        json_agg(json_build_object(
+          'oid', r_opiskeluoikeus.opiskeluoikeus_oid,
+          'tyyppi', json_build_object(
+            'koodiarvo', r_opiskeluoikeus.koulutusmuoto,
+            'koodistoUri', 'opiskeluoikeudentyyppi'
+          ),
+          'oppilaitos', json_build_object(
+            'oid', r_opiskeluoikeus.oppilaitos_oid,
+            'nimi', json_build_object(
+              'fi', r_opiskeluoikeus.oppilaitos_nimi
+            )
+          ),
+          'alkamispäivä', r_opiskeluoikeus.alkamispaiva,
+          'päättymispäivä', r_opiskeluoikeus.paattymispaiva,
+          'ryhmä', r_opiskeluoikeus.luokka
+        )) opiskeluoikeudet
+      FROM
+        r_henkilo
+        JOIN r_opiskeluoikeus ON
+          r_opiskeluoikeus.oppija_oid = r_henkilo.oppija_oid
+          AND r_opiskeluoikeus.oppilaitos_oid = any($oppilaitosOids)
+        JOIN r_paatason_suoritus ON r_paatason_suoritus.opiskeluoikeus_oid = r_opiskeluoikeus.opiskeluoikeus_oid
+        -- Lasketaan voimassaolevien kotiopetusjaksojen määrä ehtoa 4a varten
+        CROSS JOIN LATERAL (
+          SELECT count(*) AS count
+          FROM jsonb_array_elements(r_opiskeluoikeus.data -> 'lisätiedot' -> 'kotiopetusjaksot') jaksot
+          WHERE jaksot ->> 'loppu' IS NULL
+             OR to_char(NOW(), 'YYYY-MM-DD') BETWEEN jaksot ->> 'alku' AND jaksot ->> 'loppu'
+        ) kotiopetusjaksoja
+      WHERE
+        -- (1) oppija on potentiaalisesti oppivelvollinen, eli syntynyt 2004 tai myöhemmin
+        EXTRACT(YEAR FROM r_henkilo.syntymaaika) >= 2004
+        -- (2) oppijalla on Koskessa peruskoulun valvojan käyttöoikeuksiin kuuluvassa organisaatiossa peruskoulun opiskeluoikeus
+        AND r_opiskeluoikeus.koulutusmuoto = 'perusopetus'
+        AND r_paatason_suoritus.suorituksen_tyyppi = 'perusopetuksenvuosiluokka'
+        -- (3) kyseisessä opiskeluoikeudessa on yhdeksännen luokan suoritus.
+        AND r_paatason_suoritus.koulutusmoduuli_koodiarvo = '9'
+        -- (4a) valvojalla on oppilaitostason oppilaitosoikeus ja opiskeluoikeuden lisätiedoista ei löydy kotiopetusjaksoa, joka osuu tälle hetkelle
+        AND kotiopetusjaksoja.count = 0
+        -- (5)  opiskeluoikeus ei ole eronnut tilassa tällä hetkellä
+        AND r_opiskeluoikeus.viimeisin_tila <> 'eronnut'
+        AND (
+          -- (6a) opiskeluoikeus on läsnä tai väliaikaisesti keskeytynyt tällä hetkellä
+          r_opiskeluoikeus.viimeisin_tila = any('{lasna, valiaikaisestikeskeytynyt}')
+          -- TAI:
+          OR (
+            -- (6b.1 ) opiskeluoikeus on valmistunut-tilassa, ja siitä löytyy vahvistettu päättötodistus
```

Comment:
Vahvistettu päättötodistus -tarkistus mielestäni puuttuu. En tiedä, mitä tarkoittaa, että joku olisi valmistunut-tilassa ilman vahvistettua päättötodistusta, mutta Juho tämän vahvistetun päättötodistuksen erikseen mainitsi, kun sääntöjä speksattiin.

Reply:
Onko vahvistetun päättötodistuksen olemassaolo pääteltävissä siitä, että `r_paatason_suoritus.vahvistus_paiva IS NOT NULL`?

Reply:
En tiedä, pitää kysyä Juholta tai asiantuntijoilta. Perusopetuksessa on myös vuosiluokille rinnakkainen päätason suoritus suoritustyypillä `perusopetuksenoppimaara` https://virkailija.testiopintopolku.fi/koski/json-schema-viewer/#viewer-page?v=1-0-11-9-0-1 , saattaisi viitata myös sen olemassoloon.

Reply:
Kysyin Juholta, tuon vahvistettu päättötodistus-checkin voi poistaa kommentista eikä tarvitse toteuttaa, se on redundantti.